ntoskrnl/se/token.c

```diff
@@ -1856,7 +1856,10 @@ SepCreateSystemProcessToken(VOID)
 
     Privileges[i].Attributes = 0;
     Privileges[i++].Luid = SeSystemtimePrivilege;
-    ASSERT(i == 20);
+
+    Privileges[i].Attributes = SE_PRIVILEGE_ENABLED_BY_DEFAULT | SE_PRIVILEGE_ENABLED;
+    Privileges[i++].Luid = SeImpersonatePrivilege;
+    ASSERT(i == 21);
```

Comment:
Oh, this is just a statically initialized array. Please, do it the proper way:
```
LUID_AND_ATTRIBUTES Privileges[] = {
{
    .Luid = SeTcbPrivilege,
    .Attributes = SE_PRIVILEGE_ENABLED_BY_DEFAULT | SE_PRIVILEGE_ENABLED
},
{
    .Luid = SeCreateTokenPrivilege,
},
{
    .Luid = SeTakeOwnershipPrivilege,
},
...
};
```
and then when you need to take the amount of it, just use `RTL_NUMBER_OF(Privileges)`

Reply:
Please do not artificially worsen Georges code by giving recommendations extravert34 that will break the compatibility of essential ros components like the kernel to older compilers with your C99-isms. E.g. VS2010 will not accept that way of initializing the struct, which you suggested. Neither in a .cpp file nor in a .c file. Your suggestion is very unportable code with poor compiler compatibility.

That RTL_NUMBER_OF(Priviliges) part is unaffected by my comment, therefore I did not resolve this conversation yet.

Reply:
@JoachimHenze would you stop repeating that bullshit about MSVC2010 and "not portable code" and finally make your own project with VC6 and Turbo C++ support?

Why should everyone satisfy you just because of your stupid arguments?
Instead of writing clean and correct code, you're trying to sabotage the project for the sake of "compatibility".